src/libraries/System.Private.CoreLib/src/System/Memory.cs

```diff
@@ -343,7 +343,7 @@ public unsafe Span<T> Span
                     }
 #endif
 
-                    refToReturn = ref Unsafe.Add(ref refToReturn, (IntPtr)(void*)desiredStartIndex);
+                    refToReturn = ref Unsafe.Add(ref refToReturn, (nint)desiredStartIndex);
```

Comment:
How come we need to cast to `nint` in the first place? Doesn't `Unsafe.Add` have a `nuint` overload now? 🤔

Reply:
Seems to be a pattern in `System.Private.CoreLib`. Possibly `Add<T>(T, UIntPtr)` is not handled in an intrinsic on some platforms?

Reply:
It is more likely left-over from the days when IntPtr/UIntPtr were not regular integers in C#.